### ISSUE TITLE
mayastor addon does not start correctly on arm64 platform

### ISSUE THREAD

**Issue by marosg42** [canonical/microk8s-core-addons#180]
#### Summary

When enabling mayastor addon on arm64 servers all pods get to running state except `rest` pod which stays in CrashLoopBackoff. If I monitor `get pod -A` I can see OOMKilled status from time to time.
Pools are in error state. 
It is arm64 only issue, the same setup works fine on amd64.

#### What Should Happen Instead?

All pods should be in Running state and pools should be created.

#### Reproduction Steps

Yes, I can, but only on arm64.
Build a six node microk8s cluster, configure prerequisites as described in docs, enable mayastor addon.

#### Introspection Report

[inspection-report-20230413_124122.tar.gz](https://github.com/canonical/microk8s-core-addons/files/11222279/inspection-report-20230413_124122.tar.gz)

**Comment by neoaggelos** [comment]
Hi @marosg42, can you manually increase the resource limits to see what the cap is for the rest pod? `microk8s kubectl edit` can help you with this.

I have not been able to reproduce this issue, can you start doubling the limits until the pod does not crash? If it's just 1-2 times more, then I guess we can bump the limits, otherwise we would have to investigate further for possible memory leaks and/or a worse bug.

Thanks!

**Comment by marosg42** [comment]
@neoaggelos I could not reproduce it on new 1.27 but 1.26 still gives the error
I will leave it up to you if you close the issue, I am on with using 1.27

I tried to change limits out of curiosity but I got
```
# pods "rest-77d69fb479-x8zqj" was not valid:
# * spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds`, `spec.tolerations` (only additions to existing tolerations) or `spec.terminationGracePeriodSeconds` (allow it to be set to 1 if it was previously negative)
```

**Comment by neoaggelos** [comment]
Yes, you should not be changing the pods themselves, but rather the spec template for the deployment, with:

```
microk8s kubectl edit deploy/rest -n mayastor
```

Apologies for not making it clear in the original message!

**Comment by marosg42** [comment]
on 1.26 with memory limit set to 512Mi it was still crashing, after switching to 1Gi it kept running.
